Zpp/Simulation/Messages/MaterialRequest.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Zpp.Simulation.Messages
+{
+    public class MaterialRequest : IComparable<MaterialRequest>
+    {
+        private static int count = 1;
+        public MaterialRequest(Material material, Dictionary<int, bool> childRequests,int parrent, long due, bool isHead)
+        {
+            Id = GetCounter();
+            Material = material;
+            Parrent = parrent;
+            ChildRequests = childRequests;
+            Due = due;
+            IsHead = isHead;
+        }
+
+        static int GetCounter()
+        {
+            return count++;
+        }
+        public bool IsHead { get; }
+        public int Id { get;  }
+        public Material Material { get;  }
+        public int Parrent { get; }
+        public Dictionary<int, bool> ChildRequests { get; }
+        public long Due { get; }
```

Comment:
I know there is no sorting rule in C#, but usually all members/properties are on top, then all constructors and then at the end all methods.

Reply:
yep. Changed.